From 2d0d2708cf82621bbea20ab91e5d72604ace04be Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 3 May 2016 15:58:46 +0800
Subject: [PATCH 1197/5242] MLK-12928-14 mfd: pf1550: add otp read interface

commit  1cc345be7598f43717997427801073cc2dc2ebfd from
https://source.codeaurora.org/external/imx/linux-imx.git

add otp read interface to let pf1550 regulator driver or other sub driver
to read out otp register.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 include/linux/mfd/pf1550.h |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/linux/mfd/pf1550.h b/include/linux/mfd/pf1550.h
index f19c493..76f0ae8 100644
--- a/include/linux/mfd/pf1550.h
+++ b/include/linux/mfd/pf1550.h
@@ -121,6 +121,10 @@ enum pf1550_pmic_reg {
 	PF1550_CHARG_REG_LED_CNFG		= 0x9E,
 	PF1550_CHARG_REG_CHGR_KEY2		= 0x9F,
 
+	PF1550_TEST_REG_FMRADDR			= 0xC4,
+	PF1550_TEST_REG_FMRDATA			= 0xC5,
+	PF1550_TEST_REG_KEY3			= 0xDF,
+
 	PF1550_PMIC_REG_END			= 0xff,
 };
 
@@ -241,4 +245,7 @@ struct pf1550_dev {
 	int irq;
 };
 
+int pf1550_read_otp(struct pf1550_dev *pf1550, unsigned int index,
+			unsigned int *val);
+
 #endif /* __LINUX_MFD_PF1550_H */
-- 
1.7.9.5

